lib/configure.js

```diff
@@ -233,16 +233,45 @@ function configure (gyp, argv, callback) {
                         'out/Release/node.exp',
                         'out/Debug/node.exp',
                         'node.exp']
+      var found = false
+      var prefix = 'find exports file'
       for (var next = 0; next < candidates.length; next++) {
          node_exp_file = path.resolve(node_root_dir, candidates[next])
          try {
-           fs.accessSync(node_exp_file, fs.R_OK)
-           // exp file found, stop looking
-           break
+           var stats = fs.statSync(node_exp_file)
+           if (stats.uid === process.getuid()) {
+             if (stats.mode & 0400) {
+               log.silly(prefix, '%s has owner read permission', node_exp_file)
+               found = true
+               break
+             }
+           } else if (process.getgroups().some(function (group) {
+               return stats.gid === group
+             }, false)) {
+             if (stats.mode & 0040) {
+               log.silly(prefix, '%s has group read permission', node_exp_file)
+               found = true
+               break
+             }
+           } else {
+             if (stats.mode & 0004) {
+               log.silly(prefix, '%s has others read permission', node_exp_file)
+               found = true
+               break
+             }
+           }
+           log.silly(prefix, 
+             '%s exists but has no read permissions', node_exp_file)
          } catch (exception) {
            // this candidate was not found or not readable, do nothing
+           log.silly(prefix, '%s does not exist', node_exp_file)
          }
       }
+      if (found) {
+        log.verbose(prefix, 'Found exports file: %s', node_exp_file)
+      } else {
+        log.error(prefix, 'Could not find exports file')
+      }
     }
```

Comment:
Can you break out this block into a separate function?

Also, the try body should be smaller.  It should basically just be the `fs.statSync()` call, nothing more, else there is too much space for bugs to hide.